webapp/javascript/components/ExportData.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBars } from "@fortawesome/free-solid-svg-icons/faBars";
+
+import clsx from "clsx";
+import { jsPDF as JSPDF } from "jspdf";
+import "jspdf-autotable";
+
+function ExportData() {
+  const [toggleMenu, setToggleMenu] = useState(false);
+
+  const formattedDate = () => {
+    const cd = new Date();
+    const d = cd.getDate() < 10 ? `0${cd.getDate()}` : `${cd.getDate()}`;
+    const m = cd.getMonth() < 10 ? `0${cd.getMonth()}` : `${cd.getMonth()}`;
+    const y = cd.getFullYear();
+    return `${d}_${m}_${y}`;
+  };
+
+  // export flamegraph canvas element
+  const exportCanvas = (mimeType) => {
+    if (mimeType === "pdf") {
+      const canvas = document.querySelector(".flamegraph-canvas");
+      const myImage = canvas.toDataURL("image/jpeg,1.0");
+
+      const pdf = new JSPDF("p", "mm", "a4");
+      const pdfWidth = pdf.internal.pageSize.getWidth();
+
+      // reduce canvas width to a fit on the pdf
+      // could use a more robust validation/assignment
+      // of the canvas width and heights
+      let count = 0;
+      let canvasWidth = canvas.width;
+      if (canvasWidth > pdfWidth) {
+        while (canvasWidth > pdfWidth) {
+          count += 1;
+          canvasWidth -= pdfWidth;
+        }
+        count += 1;
+      }
+
+      // alignment
+      const pdfXOffset = (pdfWidth - canvas.width / count) / 2;
+      const pdfYOffset = 30;
+      pdf.addImage(
+        myImage,
+        "JPEG",
+        pdfXOffset,
+        pdfYOffset,
+        canvas.width / count,
+        canvas.height / count
+      );
+
+      const textXOffset = pdfXOffset;
+      const textYOffset = 10;
+      pdf.text(textXOffset, textYOffset, "Flamegraph Visual");
```

Comment:
I think "Flamegraph Visual" could be more descriptive here. Could you create a function similar to this:
```
formatPdfTitle = () =>{ }
```

Where the function returns a title in the format:
```
{application_name} - from: {from} until: {until}
```

This will be more useful if you want to recreate the same flamegraph you can use these parameters to do so. I know sometimes the parameters will be named kind of funny (i.e. `from=now-1h` or `from=10299489403094`) but I think thats fine.